src/test/java/seedu/address/logic/commands/GetEditIngredientCommandTest.java

```diff
@@ -0,0 +1,111 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.showEditIngredientCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showIngredientAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_INGREDIENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_INGREDIENT;
+import static seedu.address.testutil.TypicalIngredients.getTypicalWishfulShrinking;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.recipe.Ingredient;
+
+public class GetEditIngredientCommandTest {
+    private Model model = new ModelManager(getTypicalWishfulShrinking(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Ingredient ingredientToEdit =
+                model.getFilteredIngredientList().get(INDEX_FIRST_INGREDIENT.getZeroBased());
+        GetEditIngredientCommand getEditIngredientCommand = new GetEditIngredientCommand(INDEX_FIRST_INGREDIENT);
+
+        String expectedMessage = String.format(GetEditIngredientCommand.MESSAGE_GET_EDIT_INGREDIENT_SUCCESS,
+                ingredientToEdit.toString());
+
+        ModelManager expectedModel = new ModelManager(model.getWishfulShrinking(), new UserPrefs());
+
+        showEditIngredientCommandSuccess(getEditIngredientCommand, model, expectedMessage, ingredientToEdit,
+                expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredIngredientList().size() + 1);
+        GetEditIngredientCommand getEditIngredientCommand = new GetEditIngredientCommand(outOfBoundIndex);
+
+        assertCommandFailure(getEditIngredientCommand, model, Messages.MESSAGE_INVALID_INGREDIENT_DISPLAYED_INDEX);
+    }
+
+    //@Test
```

Comment:
Same question with previous test

Reply:
I commented it out because not sure if its relevant. It looks like its testing the model and this command does not modify model instead it modifies CommandResult which has been tested.